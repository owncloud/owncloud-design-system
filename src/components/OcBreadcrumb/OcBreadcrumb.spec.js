import { shallowMount } from "@vue/test-utils"
import Breadcrumb from "./OcBreadcrumb.vue"

const items = [
  { text: "First folder", to: { path: "folder" } },
  { text: "Subfolder", onClick: () => alert("Breadcrumb clicked!") },
  { text: "Deep", to: { path: "folder" } },
  { text: "Deeper ellipsize in responsive mode" },
]

describe("OcBreadcrumb", () => {
  it("sets correct variation", () => {
    const wrapper = shallowMount(Breadcrumb, {
      propsData: {
        variation: "lead",
        items,
      },
    })

    expect(wrapper.props().variation).toMatch("lead")
    expect(wrapper.classes()).toContain("oc-breadcrumb-lead")
    expect(wrapper).toMatchSnapshot()
  }),
    it("displays all items", () => {
      const wrapper = shallowMount(Breadcrumb, {
        propsData: {
          items,
        },
      })

      expect(wrapper.findAll(".oc-breadcrumb-list-item").length).toBe(4)
      expect(wrapper).toMatchSnapshot()
    })
})
