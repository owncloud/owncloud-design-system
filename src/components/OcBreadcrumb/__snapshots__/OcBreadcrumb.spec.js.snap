// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcBreadcrumb displays all items 1`] = `
<nav class="oc-breadcrumb oc-breadcrumb-default">
  <ol class="oc-breadcrumb-list oc-flex">
    <li class="oc-breadcrumb-list-item oc-flex oc-flex-middle">
      <router-link-stub tag="a" to="[object Object]"><span>First folder</span></router-link-stub>
      <oc-icon-stub name="arrow-right-s" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="var(--oc-color-text-default)" class="oc-mx-xs"></oc-icon-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item oc-flex oc-flex-middle">
      <!---->
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium"><span>Subfolder</span></oc-button-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item oc-flex oc-flex-middle">
      <router-link-stub tag="a" to="[object Object]"><span>Deep</span></router-link-stub>
      <oc-icon-stub name="arrow-right-s" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="var(--oc-color-text-default)" class="oc-mx-xs"></oc-icon-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item oc-flex oc-flex-middle">
      <!----> <span aria-current="page" tabindex="-1">Deeper ellipsize in responsive mode</span>
      <!---->
    </li>
  </ol>
  <div class="oc-breadcrumb-drop"><label tabindex="0" class="oc-breadcrumb-drop-label"><span aria-current="page" class="oc-breadcrumb-drop-label-text">Deeper ellipsize in responsive mode</span>
      <oc-icon-stub name="arrow-down-s" filltype="fill" accessiblelabel="Expand more" type="span" size="medium" variation="passive" color="" class="oc-breadcrumb-drop-label-icon"></oc-icon-stub>
    </label>
    <oc-drop-stub dropid="oc-drop-2" toggle="" position="bottom-start" mode="click" paddingsize="medium">
      <ol>
        <li>
          <router-link-stub tag="a" to="[object Object]">
            Deep
          </router-link-stub>
        </li>
        <li>
          <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="left" gapsize="medium">
            Subfolder
          </oc-button-stub>
        </li>
        <li>
          <router-link-stub tag="a" to="[object Object]">
            First folder
          </router-link-stub>
        </li>
      </ol>
    </oc-drop-stub>
  </div>
</nav>
`;

exports[`OcBreadcrumb sets correct variation 1`] = `
<nav class="oc-breadcrumb oc-breadcrumb-lead">
  <ol class="oc-breadcrumb-list oc-flex">
    <li class="oc-breadcrumb-list-item oc-flex oc-flex-middle">
      <router-link-stub tag="a" to="[object Object]"><span>First folder</span></router-link-stub>
      <oc-icon-stub name="arrow-right-s" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="var(--oc-color-text-default)" class="oc-mx-xs"></oc-icon-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item oc-flex oc-flex-middle">
      <!---->
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium"><span>Subfolder</span></oc-button-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item oc-flex oc-flex-middle">
      <router-link-stub tag="a" to="[object Object]"><span>Deep</span></router-link-stub>
      <oc-icon-stub name="arrow-right-s" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="var(--oc-color-text-default)" class="oc-mx-xs"></oc-icon-stub>
      <!---->
    </li>
    <li class="oc-breadcrumb-list-item oc-flex oc-flex-middle">
      <!----> <span aria-current="page" tabindex="-1">Deeper ellipsize in responsive mode</span>
      <!---->
    </li>
  </ol>
  <div class="oc-breadcrumb-drop"><label tabindex="0" class="oc-breadcrumb-drop-label"><span aria-current="page" class="oc-breadcrumb-drop-label-text">Deeper ellipsize in responsive mode</span>
      <oc-icon-stub name="arrow-down-s" filltype="fill" accessiblelabel="Expand more" type="span" size="medium" variation="passive" color="" class="oc-breadcrumb-drop-label-icon"></oc-icon-stub>
    </label>
    <oc-drop-stub dropid="oc-drop-1" toggle="" position="bottom-start" mode="click" paddingsize="medium">
      <ol>
        <li>
          <router-link-stub tag="a" to="[object Object]">
            Deep
          </router-link-stub>
        </li>
        <li>
          <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="left" gapsize="medium">
            Subfolder
          </oc-button-stub>
        </li>
        <li>
          <router-link-stub tag="a" to="[object Object]">
            First folder
          </router-link-stub>
        </li>
      </ol>
    </oc-drop-stub>
  </div>
</nav>
`;
